src/Controller/CommandLineController.ts

```diff
@@ -0,0 +1,47 @@
+import yargs from 'yargs'
+
+export interface JunknetArguments {
+	makefile: string
+	docker_image: string
+	target: string | null
+}
+
+/**
+ * Interpret a list of arguments, returning the relevant ones in the f
+ *
+ * @param argv - the invoking process's command-line arguments.
+ * @returns JunknetInterface object containing option values.
+ */
+export function interpretArgv(argv: readonly string[]): JunknetArguments {
+	// Use only the first two elements; Node.js appends extra elements to process.argv.
+	const yargsArgv = yargs(argv.slice(2)).options({
```

Comment:
It would be better to assume the caller has provided what we actually want to parse. Say the arguments don't come from `process.argv` originally, you don't want to make them do something silly like prepend two placeholder items.